dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -151,59 +176,34 @@ class DdAndroidGradlePlugin @Inject constructor(
     }
 
     @Suppress("StringLiteralDuplication")
-    internal fun configureBuildIdGenerationTask(
+    private fun configureBuildIdGenerationTask(
         target: Project,
-        appExtension: AppExtension,
-        variant: ApplicationVariant
+        variant: AppVariant
     ): TaskProvider<GenerateBuildIdTask> {
         val buildIdDirectory = target.layout.buildDirectory
             .dir(Path("generated", "datadog", "buildId", variant.name).toString())
         val buildIdGenerationTask = GenerateBuildIdTask.register(target, variant, buildIdDirectory)
 
-        // we could generate buildIdDirectory inside GenerateBuildIdTask and read it here as
-        // property using flatMap, but when Gradle sync is done inside Android Studio there is an error
-        // Querying the mapped value of provider (java.util.Set) before task ... has completed is
-        // not supported, which doesn't happen when Android Studio is not used (pure Gradle build)
-        // so applying such workaround
-        // TODO RUM-0000 use new AndroidComponents API to inject generated stuff, it is more friendly
-        appExtension.sourceSets.getByName(variant.name).assets.srcDir(buildIdDirectory)
-
-        val variantName = variant.name.capitalize()
-        listOf(
-            "package${variantName}Bundle",
-            "build${variantName}PreBundle",
-            "lintVitalAnalyze$variantName",
-            "lintVitalReport$variantName",
-            "generate${variantName}LintVitalReportModel"
-        ).forEach {
-            target.tasks.findByName(it)?.dependsOn(buildIdGenerationTask)
-        }
-
-        // don't merge these 2 into list to call forEach, because common superclass for them
-        // is different between AGP versions, which may cause ClassCastException
-        variant.mergeAssetsProvider.configure { it.dependsOn(buildIdGenerationTask) }
-        variant.packageApplicationProvider.configure { it.dependsOn(buildIdGenerationTask) }
-
         return buildIdGenerationTask
     }
 
     @Suppress("DefaultLocale", "ReturnCount")
     internal fun configureVariantForUploadTask(
         target: Project,
-        variant: ApplicationVariant,
+        variant: AppVariant,
         buildIdGenerationTask: TaskProvider<GenerateBuildIdTask>,
         apiKey: ApiKey,
         extension: DdExtension
-    ): TaskProvider<DdMappingFileUploadTask> {
+    ): TaskProvider<MappingFileUploadTask> {
         val uploadTaskName = UPLOAD_TASK_NAME + variant.name.capitalize()
         val uploadTask = target.tasks.register(
             uploadTaskName,
-            DdMappingFileUploadTask::class.java,
+            MappingFileUploadTask::class.java,
             GitRepositoryDetector(execOps)
         ).apply {
             configure { uploadTask ->
                 @Suppress("MagicNumber")
-                if (DdTaskUtils.isGradleAbove(target, 7, 5)) {
+                if (TaskUtils.isGradleAbove(target, 7, 5)) {
```

Comment:
we're still using some magic numbers here, maybe we should move this in a static constant in the `CurrentAgpVersion` ?